autoload/build.vim

```diff
@@ -262,7 +365,7 @@ endfunction " }}}
 function! s:get_first_build_system_in_dir(dir, build_system_names) " {{{
   for l:build_name in a:build_system_names
     for l:build_file in split(s:get_buildsys_item(l:build_name, 'file'), ',')
-      if filereadable(a:dir . '/' . l:build_file)
+      if filereadable(a:dir . '/' . l:build_file) || !empty(glob(a:dir . '/' . l:build_file))
```

Comment:
Again it was too much anticipation.
There is the case of Chicken Scheme *chicken-install* which is effectively a build system and use a `name_of_project.egg` file.
Since the change to support glob was so simple it seemed ok to me.
What kind of magic is a problem to you ?

Reply:
I see the need to support `*.egg`. But I still have mixed feelings. Glob allows recursive searches via `glob('**')` and even run shell commands:

```vim
glob('`find -iname "*.cpp"`')
```

By default its behaviour is dependent on `wildignore` and `suffixes`.

Reply:
What if `a:dir` contains special characters? E.g. the PWD is `/home/user/**`, which is a legal directory name, but gets interpreted by `glob()`. It could be worked around by lchdir + `glob('.' ...)` but thats hacky.